cpp/src/arrow/flight/sql/odbc/CMakeLists.txt

```diff
@@ -17,5 +17,10 @@
 
 add_custom_target(arrow_flight_sql_odbc)
 
+# Use C++ 20 for ODBC and its subdirectory
+# GH-44792: Arrow will switch to C++ 20
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
This change is duplicated at https://github.com/apache/arrow/pull/47517. But it is required to fix the build. After this PR is merged, I will remove duplication at https://github.com/apache/arrow/pull/47517